src/collection.jl

```diff
@@ -0,0 +1,100 @@
+```julia
+using Glob
+fitscollection(...; exclude=fn"tek001*.fits") # same as regex match above
+```
+Similarly, `exclude_dir` allows excluding entire folders using pattern matching (e.g. skipping a backup folder `exclude_dir="backup"`).
+`exclude_key` allows excluding certain entries in the header unit of `ImageHDU` in FITS files (e.g. skipping `"HISTORY"` and `""` `exclude_key = ("HISTORY", "")`).
+
+For more information about the file matching and path deconstruction, see the extended help (`??fitscollection`)
+# Extended Help
+
+## Parts of a path
+
+Here is an example of a file path and how it would be parsed
+
+```
+     root         dir   base   ext
+[---------------][----][-----][----]
+\Users\miles\data\test\tek0001.fits
+
+     root          dir     base    ext
+[---------------][--------][-----][----]
+\Users\miles\data\test\sci\tek0001.fits
+```
+
+If `keepext` is `true`, `name=base * ext`, otherwise it is just `base`. If `abspath` is `true`, the path will be `root * dir * base * ext`, otherwise it will be `dir * base * ext`. These options allow flexility in creating a table that can be easily saved and loaded to avoid having to manually filter files. Especially consider how `abspath` can allow keeping tables that will transfer easily between computers or between data sources with common structures.
+"""
+function fitscollection(basedir::String;
+                        recursive = true,
+                        abspath = true,
+                        keepext = true,
+                        ext = r"fits(\.tar\.gz)?"i,
+                        exclude = nothing,
+                        exclude_dir = nothing,
+                        exclude_key = ("", "HISTORY"))
+    df = DataFrame()
+
+    for (root, dirs, files) in walkdir(basedir)
+        # recursive searching functionality
+        recursive || root == basedir || continue
+        # To exclude certain directories
+        if exclude_dir !== nothing
+            occursin(exclude_dir, root) && continue
+        end
+        for filename in files
+            # accept file if .fits or .fits.tar.gz
+            match_extension(filename, ext) || continue
+            # excluding the files specified by user
+            if exclude !== nothing
+                occursin(exclude, filename) && continue
+            end
+            location = joinpath(root, filename)
+            fits_data = FITS(location)
+
+            for (index, hdu) in enumerate(fits_data)
+                hdu isa ImageHDU || continue
+                header_data = read_header(hdu)
+                path = abspath ? Base.abspath(location) : location
+                name = keepext ? filename : first(split(filename, ext))[1:end-1]
+
+                # filtering out comment columns
+                _keys = filter(k -> k ∉ exclude_key, keys(header_data))
+                _values = map(_keys) do k
+                    header_data[k]
+                end
```

Comment:
this is a style thing more than anything, but I don't feel like we need the `do` syntax here. I would have simply done
```julia
_values = (header_data[k] for k in _keys)
```